File: OPSWAT-SDK/extract/windows/docs/sample_code/sample-code-cs/sample-code-cs/OesisDataLossProtection.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Text;

namespace sample_code_cs
{
    class OesisDataLossProtection : OesisGenericProduct
    {
        private static readonly OesisDataLossProtection m_instance = new OesisDataLossProtection();

        public OesisDataLossProtection() { }

        public static OesisDataLossProtection instance
        {
            get
            {
                return m_instance;
            }
        }

        public int InstallDLPModule(int methodId, ref JObject result)
        {
            string json_in = "";
            Dictionary<string, object> input = new Dictionary<string, object>()
            {
                { "method", methodId },
            };
            OesisUtils.CreateJsonIn(ref json_in, input);
            string json_out;
            int rc = this.oesis.Invoke(json_in, out json_out);
            OesisUtils.GetJsonValues(ref result, rc, json_out, new List<string> { });

            return rc;
        }

        public int UninstallDLPModule(int methodId, ref JObject result)
        {
            string json_in = "";
            Dictionary<string, object> input = new Dictionary<string, object>()
            {
                { "method", methodId },
            };
            OesisUtils.CreateJsonIn(ref json_in, input);
            string json_out;
            int rc = this.oesis.Invoke(json_in, out json_out);
            OesisUtils.GetJsonValues(ref result, rc, json_out, new List<string> { });
            return rc;
        }

        public int ManageDLPModule(ref JObject result, int methodId, string operation = "")
        {
            string json_in = "";
            Dictionary<string, object> input = new Dictionary<string, object>()
            {
                { "method", methodId }
            };
            if(operation != "")
            {
                input.Add("operation", operation);
            }
            OesisUtils.CreateJsonIn(ref json_in, input);
            string json_out;
            int rc = this.oesis.Invoke(json_in, out json_out);
            OesisUtils.GetJsonValues(ref result, rc, json_out, new List<string> { });
            return rc;
        }

        public int GetDLPState(int methodId, ref JObject result)
        {
            string json_in = "";
            Dictionary<string, object> input = new Dictionary<string, object>()
            {
                { "method", methodId }
            };
            OesisUtils.CreateJsonIn(ref json_in, input);
            string json_out;
            int rc = this.oesis.Invoke(json_in, out json_out);
            OesisUtils.GetJsonValues(ref result, rc, json_out, new List<string> { "state", "enabled" });
            return rc;
        }
    }
}

File: OPSWAT-SDK/extract/windows/docs/sample_code/sample-code-py/sample-code-py/oesis_data_loss_protection.py
from oesis_framework import *

class OesisDataLossProtection:

    def __init__(self, oesis):
        self.oesis = oesis

    def setup_dlp_module(self, method_id):
        input = {
            "method": method_id
        }
        rc, json_out = self.oesis.invoke_method(input)
        output = OesisUtils.get_json_value(rc, json_out, [])
        return rc, output
    
    def manage_dlp_module(self, method_id, operation = ''):
        input = {
            "method": method_id
        }
        if operation != '':
            input['operation'] = operation
        rc, json_out = self.oesis.invoke_method(input)
        ouput = OesisUtils.get_json_value(rc, json_out, [])
        return rc, ouput

    def get_dlp_state(self, method_id):
        input = {
            "method": method_id
        }
        rc, json_out = self.oesis.invoke_method(input)
        ouput = OesisUtils.get_json_value(rc, json_out, ["state", "enabled"])
        return rc, ouput